Content.Server/Magic/MagicSystem.cs

```diff
@@ -372,5 +385,47 @@ private void SpawnSpellHelper(List<EntitySpawnEntry> entityEntries, EntityCoordi
         }
     }
 
-    #endregion
+    private void OnGateSpell(GateSpellEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var spawnLocations = EntityManager.EntityQuery<VentCritterSpawnLocationComponent>().ToList();
+        var spawnAmount = (args.SpawnAmount);
+        _random.Shuffle(spawnLocations); ;
+
+        if (Deleted(args.Performer))
+            return;
+
+        // Create the first portal.
+        if (args.Handled = true)
+        {
+
+            foreach (var location in spawnLocations)
+            {
+
+                if (spawnAmount-- == 0)
+                    break;
+
+                var xform = Transform(args.Performer);
+                if (xform.ParentUid != xform.GridUid)
+                    return;
+                {
+                    var timeout = EnsureComp<PortalTimeoutComponent> (args.Performer);
+                    timeout.EnteredPortal = null;
+                    var coords = Transform(location.Owner);
+                    args.FirstPortal = Spawn(args.FirstPortalPrototype, coords.Coordinates);
+                    args.SecondPortal = Spawn(args.SecondPortalPrototype, Transform(args.Performer).Coordinates);
+                    _adminLogger.Add(LogType.EntitySpawn, LogImpact.Low, $"{ToPrettyString(args.Performer):player} opened {ToPrettyString(args.SecondPortal.Value)} at {Transform(args.SecondPortal.Value).Coordinates} linked to {ToPrettyString(args.FirstPortal!.Value)} using {ToPrettyString(args.Performer)}");
+                    _link.TryLink(args.FirstPortal!.Value, args.SecondPortal.Value, true);
+                    _audio.PlayPvs(args.NewPortalSound, args.Performer);
+                    args.CooldownEndTime = _timing.CurTime + args.CooldownLength;
+                    args.Handled = true;
+                }
+
```

Comment:
This looks like it was copied from vent critter code, just put it somewhere common as it would be in 4-5 areas of the codebase.

Gate also isn't a good description for what this spell does.

Also this null suppression looks extremely suss.